sklearn/ensemble/gradient_boosting.py

```diff
@@ -1417,11 +1420,21 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # init state
             self._init_state()
 
-            # fit initial model - FIXME make sample_weight optional
-            self.init_.fit(X, y, sample_weight)
+            # fit initial model
+            try:
+                self.init_.fit(X, y, sample_weight=sample_weight)
+            except TypeError:
```

Comment:
Maybe calling `inspect.signature(self.init_.fit)`, and adapting the kwargs, in consequence, would be cleaner? Here I wonder what happens if one gets a genuine unrelated TypeError, then understanding the traceback will be somewhat difficult.

Reply:
Nevermind, I see, it was the case originally and Joel requested the opposite change in  https://github.com/scikit-learn/scikit-learn/pull/12436#discussion_r228739376